coffee/base.py

```diff
@@ -969,25 +969,26 @@ class FunDecl(Statement):
 
     Syntax: ::
 
-        [pred] ret name ([args]) {body};
+        [template] [pred] ret name ([args]) {body};
 
     E.g.: ::
 
         static inline void foo(int a, int b) {return;};"""
 
-    def __init__(self, ret, name, args, body, pred=[], headers=None):
+    def __init__(self, ret, name, args, body, pred=[], headers=None, template=[]):
```

Comment:
Can you elaborate what you mean by "fix `pred`"? How would you rather `pred` be handled?